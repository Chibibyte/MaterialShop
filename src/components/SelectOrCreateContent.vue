<template>
  <v-text-field
    :dense="dense"
    :class="dense?'':'container'"
    required
    :rules="[rules.required]"
    :label="label"
    v-model="innerValue"
    ref="textfield"
  ></v-text-field>
</template>

<script>
export default {
  name: "SelectOrCreateContent",
  props: {
    label: String,
    value: String,
    dense: {
      type: Boolean,
      default: false
    }
  },
  data() {
    return {
      innerValue: this.$props.value,
      rules: {
        required: v => !!v || "Field is required"
      }
    };
  },
  watch: {
    value() {
      this.innerValue = this.value;
    },
    innerValue() {
      this.$emit("input", this.innerValue);
    }
  }
};
</script>