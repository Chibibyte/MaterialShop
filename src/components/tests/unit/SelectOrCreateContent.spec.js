import { shallowMount } from '@vue/test-utils';
import SelectOrCreateContent from '@/components/SelectOrCreateContent.vue';

let wrapper;
let stubs = [
    'v-text-field'
]

beforeEach(() => {
    wrapper = shallowMount(SelectOrCreateContent, { stubs });
})

// initial
describe('Component', () => {
    test('is instantiated', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    })

    test('default snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    })

    test('props: label', () => {
        expect(wrapper.vm.label).toBeUndefined();
    })

    test('props: value', () => {
        expect(wrapper.vm.value).toBeUndefined();
    })

    test('props: dense', () => {
        expect(wrapper.vm.dense).toBeFalsy();
    })

    test('data: innerValue', () => {
        expect(wrapper.vm.innerValue).toEqual(wrapper.vm.value);
    })
})

// watch
describe('watch', () => {
    test('value', async () => {
        await wrapper.setProps({ value: 'testString' });
        expect(wrapper.vm.innerValue).toEqual(wrapper.vm.value);
    })

    test('innerValue', async () => {
        wrapper.vm.innerValue = "testString2";
        await wrapper.vm.$nextTick();
        expect(wrapper.emitted().input).toBeTruthy();
        expect(wrapper.emitted().input[0][0]).toEqual(wrapper.vm.innerValue);
    })
})

