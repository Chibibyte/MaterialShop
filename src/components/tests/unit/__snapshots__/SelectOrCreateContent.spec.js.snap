// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component default snapshot 1`] = `
<v-text-field-stub
  class="container"
  required=""
  rules="v => !!v || \\"Field is required\\""
/>
`;
